consensus/src/round_manager.rs

```diff
@@ -1296,6 +1355,75 @@ impl RoundManager {
         }
     }
 
+    async fn process_timeout_reception_result(
+        &mut self,
+        timeout: &RoundTimeout,
+        result: VoteReceptionResult,
+    ) -> anyhow::Result<()> {
+        let round = timeout.round();
+        match result {
+            VoteReceptionResult::New2ChainTimeoutCertificate(tc) => {
+                self.new_2chain_tc_aggregated(tc).await
+            },
+            VoteReceptionResult::EchoTimeout(_) if !self.round_state.is_vote_timeout() => {
+                self.process_local_timeout(round).await
+            },
+            VoteReceptionResult::VoteAdded(_)
+            | VoteReceptionResult::VoteAddedQCDelayed(_)
+            | VoteReceptionResult::EchoTimeout(_) => Ok(()),
+            result @ VoteReceptionResult::NewQuorumCertificate(_)
+            | result @ VoteReceptionResult::DuplicateVote => {
+                bail!("Unexpected result from timeout processing: {:?}", result);
+            },
+            e => Err(anyhow::anyhow!("{:?}", e)),
+        }
+    }
+
+    pub async fn process_round_timeout_msg(
+        &mut self,
+        round_timeout_msg: RoundTimeoutMsg,
+    ) -> anyhow::Result<()> {
+        fail_point!("consensus::process_round_timeout_msg", |_| {
+            Err(anyhow::anyhow!(
+                "Injected error in process_round_timeout_msg"
+            ))
+        });
+        // Check whether this validator is a valid recipient of the vote.
+        if self
+            .ensure_round_and_sync_up(
+                round_timeout_msg.round(),
+                round_timeout_msg.sync_info(),
+                round_timeout_msg.author(),
+            )
+            .await
+            .context("[RoundManager] Stop processing vote")?
+        {
+            self.process_round_timeout(round_timeout_msg.timeout())
+                .await
+                .context("[RoundManager] Add a new timeout")?;
+        }
+        Ok(())
+    }
+
+    async fn process_round_timeout(&mut self, timeout: RoundTimeout) -> anyhow::Result<()> {
+        info!(
+            self.new_log(LogEvent::ReceiveRoundTimeout)
+                .remote_peer(timeout.author()),
+            vote = %timeout,
+            epoch = timeout.epoch(),
+            round = timeout.round(),
+        );
+        if !self.local_config.enable_round_timeout_msg {
```

Comment:
this too, the first release needs to have nodes enabling receiving, then the following release enabling the flag for sending